hastory-server/src/Data/Hastory/Server/Utils.hs

```diff
@@ -2,18 +2,24 @@
 
 module Data.Hastory.Server.Utils where
 
-import Control.Monad.Except
+import Control.Monad.Error.Class
 import Crypto.Hash (Digest, SHA256(..), hashWith)
 import qualified Data.ByteString as B
 import Data.Hastory.Types
 import qualified Data.Text as T
 import qualified Data.Text.Encoding as T
 import Data.Time.Format (defaultTimeLocale, formatTime, iso8601DateFormat)
 import Path (fromAbsDir)
+import Servant.Server
 
-ensureWith :: MonadError e m => e -> Maybe a -> m a
+import Data.Hastory.Server.HastoryHandler
+
+ensureWith :: Handler a -> Maybe a -> HastoryHandler a
 ensureWith _ (Just a) = pure a
-ensureWith e Nothing = throwError e
+ensureWith defaultAction Nothing = lift defaultAction
+
+ensureWithUnauthorized :: Maybe a -> HastoryHandler a
```

Comment:
Aha! Now that you've done this refactor, you've bumped into the natural way to do this.
Instead of `ensureWithUnAuthorized`, try writing this function:

```
withUser :: UserName -> (Entity User -> Handler a) -> Handler a-
```

Reply:
Ah, I like this a ton. Passing a callback and let `withUser` do the rest.